pymc_marketing/clv/distributions.py

```diff
@@ -19,59 +19,47 @@ class ContNonContractRV(RandomVariable):
     dtype = "floatX"
     _print_name = ("ContNonContract", "\\operatorname{ContNonContract}")
 
-    def make_node(self, rng, size, dtype, lam, p, T, T0):
+    def make_node(self, rng, size, dtype, lam, p, T):
         T = pt.as_tensor_variable(T)
-        T0 = pt.as_tensor_variable(T0)
 
-        return super().make_node(rng, size, dtype, lam, p, T, T0)
-
-    def __call__(self, lam, p, T, T0=0, size=None, **kwargs):
-        return super().__call__(lam, p, T, T0, size=size, **kwargs)
+        return super().make_node(rng, size, dtype, lam, p, T)
 
     @classmethod
-    def rng_fn(cls, rng, lam, p, T, T0, size) -> np.ndarray:
+    def rng_fn(cls, rng, lam, p, T, size) -> np.ndarray:
         size = pm.distributions.shape_utils.to_tuple(size)
 
         # TODO: broadcast sizes
         lam = np.asarray(lam)
         p = np.asarray(p)
         T = np.asarray(T)
-        T0 = np.asarray(T0)
 
         if size == ():
-            size = np.broadcast_shapes(lam.shape, p.shape, T.shape, T0.shape)
+            size = np.broadcast_shapes(lam.shape, p.shape, T.shape)
 
         lam = np.broadcast_to(lam, size)
         p = np.broadcast_to(p, size)
         T = np.broadcast_to(T, size)
-        T0 = np.broadcast_to(T0, size)
 
-        output = np.zeros(shape=size + (2,))
+        x_1 = rng.poisson(lam * T)
+        x_2 = rng.geometric(p)
+        x = np.minimum(x_1, x_2)
 
-        # TODO: Optimize to work in a vectorized manner!
-        def sim_data(lam, p, T, T0):
-            t = T0
-            n = 0
-
-            while True:
-                wait = rng.exponential(scale=1 / lam)
-                dropout = rng.binomial(n=1, p=p)
-
-                if t + wait > T:
-                    break
-                else:
-                    t += wait
-                    n += 1
+        def modified_rng_beta(a, b):
+            if a == 0:
+                "If N(T) = 0, then t_x = t_0 = 0."
+                return 0
 
-                    if dropout == 1:
-                        break
+            return rng.beta(a, b)
 
-            return np.array([t, n])
-
-        for index in np.ndindex(*size):
-            output[index] = sim_data(lam[index], p[index], T[index], T0[index])
+        t_x = (
```

Comment:
Are you sure this is equivalent? Could you add some comments with the derivation perhaps?

Reply:
Oh I saw your explanation on the top message :O really cool! I would suggest to add it as a comment in the code for future devs ;)

Reply:
Should I add it in the `rng_fn` docstring? What about writing it as a blogpost and linking it somewhere in the class? I think that the math can be a bit terse and pictures would complement well the derivation

Reply:
Yeah that docstring would be a good place, regardless of a neat blogpost ;)